agent/api/task_unix.go

```diff
@@ -56,7 +58,8 @@ func (task *Task) BuildCgroupRoot() (string, error) {
 		return "", errors.Wrapf(err, "task build cgroup root: unable to get task-id from task ARN: %s", task.Arn)
 	}
 
-	return filepath.Join(config.DefaultTaskCgroupPrefix, taskID), nil
+	// See https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/7/html/resource_management_guide/sec-default_cgroup_hierarchies
+	return fmt.Sprintf("%s-%s.slice", task.cgroupPrefix, strings.Replace(taskID, "-", "_", -1)), nil
```

Comment:
I'm concerned about this change since you're unconditionally using the systemd slice.  Non-systemd distributions like Amazon Linux shouldn't use this format, so I think we'd need to turn this into conditional behavior.

Reply:
Yes maybe it should. But then it should probably auto detect if it's running under systemd. I didn't implement it in the because it should work even with non systemd distros and I saw that Amazon Linux 2 is using systemd.

I can continue this after the holidays, but if this is i prior for Amazon Linux 2 feel free to take this over.

Reply:
@rickard-von-essen Which distros are enabling Docker's systemd cgroups support by default? Docker's official packaging for [RPM based](https://github.com/docker/docker-ce/blob/master/components/packaging/rpm/systemd/docker.service) and [Debian based](https://github.com/docker/docker-ce/blob/master/components/packaging/deb/systemd/docker.service) distros does not, and comments in the unit files suggest that there are issues with doing so. They don't provide any additional details about the issues, though. Given this, it's unclear that we'll enable the systemd cgroups driver when we ship our Amazon Linux 2 ECS Optimized AMI.

In any case, the [/info](https://docs.docker.com/engine/api/v1.33/#operation/SystemInfo) Docker api provides the cgroup driver details in the `CgroupDriver` key, and we should be able to use that properly autodetect the appropriate configuration. Feel free to update this PR when convenient, or we'll get to this when we can.

Reply:
> Which distros are enabling Docker's systemd cgroups support by default? 

At least RHEL/CentOS/Fedora Atomic, and expect that the normal server flavor does it too.

> /info Docker api provides the cgroup driver details in the CgroupDriver

Excellent, I'll look into enabling this automatically.